app/utils/arrays.ts

```diff
@@ -34,3 +34,9 @@ export function joinListToNaturalString(arg: string[]) {
 
   return last ? `${commaJoined} and ${last}` : commaJoined;
 }
+
+export function normalizeFormFieldArray(
+  value: undefined | null | string | string[]
```

Comment:
```suggestion
  value: unknown
```

Maybe?

Reply:
Hmm, it's quite specifically one of those values the browser sends with the form when a specific input name is repeated (though maybe not null). When typing it as unknown I would have to check the array children for being strings and handle errors and so on.